File: app/src/main/java/podtest/jiki/se/podtest/start/EpisodeListViewPresenter.java
package podtest.jiki.se.podtest.start;

import podtest.jiki.se.podtest.R;
import podtest.jiki.se.podtest.model.Episode;

public class EpisodeListViewPresenter implements EpisodeContract.Presenter {
    private EpisodeViewer episodeViewer;

    public EpisodeListViewPresenter(EpisodeViewer episodeViewer) {
        this.episodeViewer = episodeViewer;
    }

    @Override
    public void update(Episode episode, EpisodeContract.View view) {
        view.setFirstRow(episode.getTitle());
        view.setSecondRow(episode.getDescription());
        view.setThumbnail(episode.getImageurl());
        view.setIcon(R.drawable.ic_info);
        view.setTag(episode);
    }

    @Override
    public void itemClicked(Object tag, @EpisodeContract.Source int source) {
        if (source == EpisodeContract.Source.ICON_IMAGE) {
            episodeViewer.showInfo((Episode) tag);
        } else {
            episodeViewer.play((Episode) tag);
        }
    }
}

File: app/src/main/java/se/asapehrsson/podtest/EpisodeListViewPresenter.kt
package se.asapehrsson.podtest

import se.asapehrsson.podtest.data.Episode

class EpisodeListViewPresenter(private val episodeViewer: EpisodeViewer) : EpisodeContract.Presenter {
    override fun close() {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun update(episode: Episode?, view: EpisodeContract.View?) {
        if (episode != null) {
            view?.setFirstRow(episode.title)
            view?.setSecondRow(episode.description)
            view?.setThumbnail(episode.imageurl)
            view?.setTag(episode)
        }
    }

    override fun itemClicked(tag: Any, request: EpisodeContract.Request) {
        if (request == EpisodeContract.Request.SHOW_DETAILS) {
            episodeViewer.showInfo(tag as Episode)
        } else {
            episodeViewer.play(tag as Episode)
        }
    }
}
